src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncBlockBodiesRequestHandler.scala

```diff
@@ -32,6 +37,20 @@ class FastSyncBlockBodiesRequestHandler(
     cleanupAndStop()
   }
 
+  private def updateBestBlockIfNeeded(receivedHashes: Seq[ByteString]): Unit = {
+    val fullBlocks = receivedHashes.flatMap { hash =>
+      for {
+        header <- blockchain.getBlockHeaderByHash(hash)
+        _ <- blockchain.getReceiptsByHash(hash)
```

Comment:
we have just received a portion of block bodies. I want to check what is the (possibly new) highest fully downloaded block (i.e the one that we already have a receipt for) - that's what this piece does.